libvast/include/vast/detail/collect.hpp

```diff
@@ -0,0 +1,27 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/detail/generator.hpp>
+
+namespace vast::detail {
+
+/// A utility function to collect all results produced by a
+/// `vast::detail::generator<T>` into a suitable standard container.
+/// Example:
+///     auto g = vast::detail::generator<int>{};
+///     auto v = vast::detail::collect<std::vector>(g);
+template <template <class, class> class Container, class Generated>
```

Comment:
Why the limitation to two arguments, and why do you hardcode the allocator below instead of defaulting it to the template argument?

E.g., this could also be:

```
template <template <class...> class Container, class T, class Allocator = std::allocator<T>>
```

Reply:
That has the drawback that you have to write out the element type `T` if you want to change the allocator.

I've changed it now to just accept a container type, and a convenience overload for the common case of `std::vector<T>`.